packages/app/src/cli/services/logs.ts

```diff
@@ -91,17 +101,34 @@ async function prepareForLogs(commandOptions: LogsOptions): Promise<{
     ...commandOptions,
     userProvidedConfigName: commandOptions.configName,
   })
-  let developerPlatformClient = selectDeveloperPlatformClient({configuration})
-  const devContextOptions: DevContextOptions = {...commandOptions, developerPlatformClient}
-  const {storeId, remoteApp, localApp} = await ensureDevContext(devContextOptions)
+  const developerPlatformClient = selectDeveloperPlatformClient({configuration})
+  const primaryStoreFqdn = commandOptions.storeFqdns?.[0]
+  const devContextOptions: DevContextOptions = {
+    ...commandOptions,
+    storeFqdn: primaryStoreFqdn,
+    developerPlatformClient,
+  }
+  const {storeId, storeFqdn, remoteApp, localApp} = await ensureDevContext(devContextOptions)
```

Comment:
We leverage `ensureDevContext` to setup the first shop, app, and org with the exact same process that dev uses. All following shops will be handled separately as additions.